tests/matrix.rs

```diff
@@ -97,6 +99,26 @@ pub mod matrix2 {
     }
 
     #[test]
+    fn test_sum_matrix() {
+        // This doesn't work because there isn't an impl for &Matrix2<f64>.
+        // let res: Matrix2<f64> = [A, B, C].iter().sum();
+
+        // This type resolves at least, but is ugly.
+        let res: Matrix2<f64> = [A, B, C].iter().cloned().sum();
+        assert_eq!(res,
+                   Matrix2::new(5.0f64, 8.0f64,
+                                6.0f64, 11.0f64));
+    }
+
+    #[test]
+    fn test_product_matrix() {
+        // Can't figure out what type `P` is?
+        assert_eq!([A, B, C].into_iter().product(),
+                   Matrix2::new(42.0f64, 54.0f64,
+                                55.0f64, 67.0f64));
```

Comment:
No, that isn't it. the `sum` test works because of the explicit type on `res`. I don't know why it can't figure this out automatically :/ .